2025-03-05 18:17:59,136 - INFO - ✅ Using GPU device 1.
2025-03-05 18:17:59,138 - WARNING - ⚠️ Output directory 'NPZ' does not exist: /path/to/your/output/npz
2025-03-05 18:17:59,139 - INFO - 📁 Output directory 'NPY' is set to: /caefs/data/IllustrisTNG/300/ICS
2025-03-05 18:17:59,139 - WARNING - ⚠️ Output directory 'LOGS' does not exist: /path/to/your/logs
2025-03-05 18:17:59,140 - INFO - 📁 Input (npy) folder: /caefs/data/IllustrisTNG/300/ICS
2025-03-05 18:17:59,140 - INFO - 📁 Output folder: /caefs/user/mmingyeong/structura/Structura/src/results
2025-03-05 18:17:59,141 - INFO - 📂 Detected 313 .npy files in the input folder.
2025-03-05 18:17:59,142 - INFO - 📄 Example file: chunk_89.npy
2025-03-05 18:17:59,142 - INFO - ✅ X range set: 64358.89 - 71133.51 ckpc/h (Center: 100 cMpc/h, Thickness: 10 cMpc/h)
2025-03-05 18:17:59,142 - INFO - 🔹 Filtering X range: 64358.89 - 71133.51 ckpc/h (64.359 - 71.134 cMpc/h)
2025-03-05 18:17:59,142 - INFO - 🔹 Loading data...
2025-03-05 18:22:35,359 - INFO - ✅ Time taken to load all chunks: 276.21s
2025-03-05 18:22:35,529 - INFO - ✅ Data loaded successfully. Shape: (5161333, 3)
2025-03-05 18:22:35,529 - INFO - 🔹 Generating 2D histogram, saving to ../results/...
2025-03-05 18:22:35,529 - ERROR - Unhandled exception occurred:
Traceback (most recent call last):
  File "/caefs/user/mmingyeong/structura/Structura/src/example/visualization_ex.py", line 61, in <module>
    viz.plot_2d_histogram(
TypeError: Visualizer.plot_2d_histogram() missing 2 required positional arguments: 'lbox_cMpc' and 'lbox_ckpch'
